gluten-arrow/src/main/scala/org/apache/gluten/expression/InterpretedColumnarProjection.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gluten.expression
+
+import org.apache.gluten.vectorized.ArrowColumnarRow
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
+import org.apache.spark.sql.catalyst.expressions.BindReferences.bindReferences
+import org.apache.spark.sql.catalyst.expressions.aggregate.NoOp
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types.{BinaryType, DataType, StringType}
+import org.apache.spark.unsafe.types.UTF8String
+
+/**
+ * A [[ArrowProjection]] that is calculated by calling `eval` on each of the specified expressions.
+ *
+ * @param expressions
+ *   a sequence of expressions that determine the value of each column of the output row.
+ */
+class InterpretedColumnarProjection(expressions: Seq[Expression]) extends ArrowProjection {
```

Comment:
`InterpretedColumnarProjection` vs `InterpretedArrowProjection` ?

Reply:
Yes, ArrowProjection corresponds to  `InterpretedArrowProjection`, we may do a small refactor to add `ColumnarRow` interface to support write to string and binary and rename to `ColumnarProjection` and `InterpretedColumnarProjection` after we have Velox Vector Java API.